src/system/controller.h

```diff
@@ -0,0 +1,56 @@
+#ifndef _CONTROLLER_H
+#define _CONTROLLER_H
+
+#include <stdint.h>
+
+// The button values match those defined by libultra - see os_cont.h. Instead of
+// hard-coding these values here, a better approach would be to have a dedicated
+// file for each library implementation and choose the appropriate one at build
+// time.
+// NOTE: libdragon uses the same values but reversed - see joypad_buttons_t.
+typedef enum {
+    BUTTON_A       = 0x8000,
+    BUTTON_B       = 0x4000,
+    BUTTON_Z       = 0x2000,
+    BUTTON_START   = 0x1000,
+    BUTTON_UP      = 0x0800,
+    BUTTON_DOWN    = 0x0400,
+    BUTTON_LEFT    = 0x0200,
+    BUTTON_RIGHT   = 0x0100,
+    BUTTON_L       = 0x0020,
+    BUTTON_R       = 0x0010,
+    BUTTON_C_UP    = 0x0008,
+    BUTTON_C_DOWN  = 0x0004,
+    BUTTON_C_LEFT  = 0x0002,
+    BUTTON_C_RIGHT = 0x0001,
+} buttons_t;
```

Comment:
Very minor thing: the current convention in this codebase is to use singular names for structs and enums in PascalCase (e.g., `ControllerButton` and `ControllerStick`).

Reply:
Fixed in 376fa90, thanks. That also made me realised the typo I fixed in 4918cef.